### DISCUSSION TITLE
[@types/react-native-vector-icons] Property 'refs' is missing in type 'Component<IconProps, any, any>'

### DISCUSSION THREAD

**Discussion by Doublemme** [DefinitelyTyped/DefinitelyTyped#71404] [Issues with a @types package]
Running the unit tests I've got this error:
```
 src/Components/Base/BaseIcon.tsx:43:14 - error TS2786: 'Icon' cannot be used as a JSX component.
      Its instance type 'Icon' is not a valid JSX element.
        Property 'refs' is missing in type 'Component<IconProps, any, any>' but required in type 'ElementClass'.

    43             <Icon
                    ~~~~

      node_modules/@types/react/ts5.0/index.d.ts:1034:9
        1034         refs: {
                     ~~~~
        'refs' is declared here.
```

And this is the error I see on the BaseIcon.tsx component

![Screenshot 2024-12-11 at 12 36 45](https://github.com/user-attachments/assets/56cca573-8970-45e0-873f-4d83c7224efe)

I'll put down here the packages and their version that I have installed

```json
"@types/react": "^18.0.24",
"@types/react-native-vector-icons": "^6.4.12",
"@types/react-test-renderer": "^18.0.0",
```

Anyone else with this problem?

[a comment by an automated account is not shown]

**Comment by tomstevens94** [comment]
Im getting the same issue!

**Comment by geroale** [comment]
Same issue here

**Comment by NJneeraj** [comment]
Same problem

**Comment by lukhol** [comment]
I’m experiencing the same issue. The problem disappears when adding the following configuration to the `compilerOptions` section of `tsconfig.json`. Unfortunately, this change breaks the Android build for me.
```
"paths": {
  "react": [ "./node_modules/@types/react" ]
}
```

**Comment by robertying** [comment]
Looks like `@types/react@19` is being installed for `@types/react-native-vector-icons` even if v18 is specified in my package.json.

You can use your package manager's override functionality to force not using `@types/react@19` as react 19 is not supported by react native yet.

For example, for pnpm:

package.json
```json
{
  ...
  "dependencies": {
    ...
    "@types/react": "18.3.18",
  },
  "pnpm": {
    "overrides": {
      "@types/react": "$@types/react"
    }
  }
}
```

**Reply by ngxhuyhoang** [reply]
Happy new year

**Reply by agilgur5** [reply]
Yea afaict it seems like the root cause of this issue is a mismatch of `react` vs `@types/react` versions, with React 19 specifically having some [incompatible breaking changes](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#typescript-changes) and so making it more of a hard requirement to make sure the versions match.

**Comment by Magdalena1610** [comment]
I had the same issue. I updated @types/react and this helped

**Comment by chiefchief** [comment]
Adding a resolution to package.json with a lower @types/react version resolves the issue.
```json
{
  ...,
  "resolutions": {
    "@types/react-native-vector-icons/@types/react": "18.3.11"
  },
  ...
}
```
